controller/k8s/watcher_test.go

```diff
@@ -51,4 +58,34 @@ func TestWatcher(t *testing.T) {
 			t.Fatalf("Expected error, got nil")
 		}
 	})
+	t.Run("Returns error if the watcher does not initialize in the time limit", func(t *testing.T) {
+		resource := &resourceToWatchImpl{}
+		var lock sync.RWMutex
+		didRetry := false
+
+		testWatchInit := func(stop <-chan struct{}) error {
+			lock.Lock()
+			defer lock.Unlock()
+			didRetry = true
+			return errors.New("mock error")
+		}
+		stopCh := make(chan struct{}, 1)
+		watcher := newWatcher(resource, "resourcestring", testWatchInit, stopCh)
+		watcher.timeout = 1 * time.Second
```

Comment:
Related to my comment above, for this test to be meaningful, we need the give the watcher a higher timeout and assert that x number of calls were made within the timeout. I think we should aim for three calls total, which should take 2.5 seconds with the backoff you have configured. So I'd bump this timeout to 3 seconds to be safe:

```
		watcher.timeout = 3 * time.Second
```